crates/nix_rs/src/version_spec.rs

```diff
@@ -0,0 +1,204 @@
+//! Rust module defining the spec for [NixVersion]
+
+use std::{fmt, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::version::{BadNixVersion, NixVersion};
+
+/// Specifies a version requirement for Nix
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum NixVersionSpec {
+    /// Version must be greater than the specified version
+    Gt(NixVersion),
+    /// Version must be greater than or equal to the specified version
+    Gteq(NixVersion),
+    /// Version must be less than the specified version
+    Lt(NixVersion),
+    /// Version must be less than or equal to the specified version
+    Lteq(NixVersion),
+    /// Version must not equal the specified version
+    Neq(NixVersion),
+}
+
+/// A collection of version requirements
```

Comment:
```suggestion
/// Version requirement for [NixVersion]
///
/// Example:
/// ```
/// >=2.8, <2.14, 12.13.4
/// ```
```